### ISSUE TITLE
globalChannelMap variables not working correctly with S3 File Writer Destination

### ISSUE THREAD

**Issue by WallyD444** [nextgenhealthcare/connect#4669]
I have mappings for the s3 bucket and folder, as well as the filename, "AWS Access key ID" and "AWS Secret Access Key".  The variables work great for the bucket, folder  and filename, BUT I'm getting an AWS error when this channel runs:

"File Writer error
ERROR MESSAGE: Error writing file
com.amazonaws.services.s3.model.AmazonS3Exception: The AWS Access Key Id you provided does not exist in our records. (Service: Amazon S3; Status Code: 403; Error Code: InvalidAccessKeyId; Request ID: BSTRXWX20WD4NW0Q; S3 Extended Request ID: OWEVncVPZalWXRW3z82nUAxUJND2nxpWA06NTcMHx6IE5S51q/u1qetSWnJVzcqgqdyPucL+3Ko=), S3 Extended Request ID: OWEVncVPZalWXRW3z82nUAxUJND2nxpWA06NTcMHx6IE5S51q/u1qetSWnJVzcqgqdyPucL+3Ko=
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1712)"
	
HOWEVER, when I hard code the "AWS Access key ID" and "AWS Secret Access Key" in it works perfectly.

I've double checked the variables I'm using and they're fine.  I put them in a globalChannelMap in the previous destination's transformer and drag them from the 'destination mappings' portion of the destination.

We're using Mirth 3.7.1

**Comment by jonbartels** [comment]
I wonder if this is an example of this more general discussion - https://github.com/nextgenhealthcare/connect/discussions/4535

When you entered the variable in the connector did you enter it like `${WallyVar}` or `{WallyVar}` or `WallyVar` ?

**Comment by tonygermano** [comment]
> I wonder if this is an example of this more general discussion - #4535
> 
> When you entered the variable in the connector did you enter it like `${WallyVar}` or `{WallyVar}` or `WallyVar` ?

It shouldn't be related to that discussion if it works when the values are hard coded. That means mirth is looking for a string values and not a map keys.

**Comment by WallyD444** [comment]
Thank you for replying, I'm not sure why, but when I completely deleted the channel and built it again from scratch it worked, still using ${my_variable_here} format.

**Comment by WallyD444** [comment]
ok, after it worked once, it started giving the same error again: "The AWS Access Key Id you provided does not exist in our records"

I tried creating another new channel, but the new one gave the same error.  It's strange that it worked once then stopped.  I also tried using {my_variable_here} and just my_variable_here formats to no avail.

**Comment by tonygermano** [comment]
If you are setting the value in one destination to be used in the next destination for the same message, you should be using the `channelMap`, not the `globalChannelMap`. You might be running into concurrency issues.

**Comment by WallyD444** [comment]
yeah, I changed it to channelMap and it's still not working.  We receive the credentials from the Vault API, then push them to the channelMap variables.  

When pushing the credentials directly to the channelMap it works, the problem occurs when they're pulled from the Vault and put into the channelMap.  I've run logger.info on a get from the channelMap for these variables and they seem fine.  I've tried to use String(channelMap.get('access_key')), again, it 'seems' fine in the dashboard log, and mappings, but still gives the AWS error.
